File: src/io/innofang/abstract_factory/example/java/StrawberryHeartCake.java
package io.innofang.abstract_factory.example.java;

/**
 * Created by InnF on 2017/2/25.
 * 草莓爱心蛋糕
 */
public class StrawberryHeartCake extends CakeFactory{
    @Override
    public CakeCream cream() {
        return new StrawberryCream();
    }

    @Override
    public CakeStyle style() {
        return new HeartStyle();
    }
}

File: src/io/innofang/abstract_factory/example/kotlin/StrawberryHeartCake.kt
package io.innofang.abstract_factory.example.kotlin

/**
 * Created by Inno Fang on 2017/9/2.
 */
class StrawberryHeartCake : CakeFactory() {
    override fun style(): CakeStyle {
        return HeartStyle()
    }

    override fun cream(): CakeCream {
        return StrawberryCream()
    }
}